cmd/clusterctl/client/cluster/mover.go

```diff
@@ -102,6 +110,100 @@ func (o *objectMover) Move(namespace string, toCluster Client, dryRun bool) erro
 	return o.move(objectGraph, proxy)
 }
 
+func (o *objectMover) Save(namespace string, directory string) error {
+	log := logf.Log
+	log.Info("Performing save...")
+
+	objectGraph := newObjectGraph(o.fromProxy, o.fromProviderInventory)
+
+	// Gets all the types defines by the CRDs installed by clusterctl plus the ConfigMap/Secret core types.
+	err := objectGraph.getDiscoveryTypes()
+	if err != nil {
+		return err
+	}
+
+	// Discovery the object graph for the selected types:
+	// - Nodes are defined the Kubernetes objects (Clusters, Machines etc.) identified during the discovery process.
+	// - Edges are derived by the OwnerReferences between nodes.
+	if err := objectGraph.Discovery(namespace); err != nil {
+		return err
+	}
+
+	// Checks if Cluster API has already completed the provisioning of the infrastructure for the objects involved in the save operation.
+	// This is required because if the infrastructure is provisioned, then we can reasonably assume that the objects we are saving are
+	// not currently waiting for long-running reconciliation loops, and so we can safely rely on the pause field on the Cluster object
+	// for blocking any further object reconciliation on the source objects.
+	if err := o.checkProvisioningCompleted(objectGraph); err != nil {
+		return err
+	}
+
+	// Check whether nodes are not included in GVK considered for move
+	objectGraph.checkVirtualNode()
+
+	if err := o.save(objectGraph, directory); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *objectMover) Restore(namespace string, glob string, directory string) error {
```

Comment:
We should have a test for this func.
Given that this is the reverse too the save test, you can eventually combine the two tests, but I leave this to you